askama_parser/src/expr.rs

```diff
@@ -74,15 +77,45 @@ pub enum Expr<'a> {
 impl<'a> Expr<'a> {
     pub(super) fn arguments(i: &'a str, level: Level) -> ParseResult<'a, Vec<Self>> {
         let (_, level) = level.nest(i)?;
+        let mut named_arguments = HashSet::new();
```

Comment:
I don't think this logic should live in `Expr::arguments()`, which is not just used for macros.

Reply:
It allows to greatly improve the parser error, but as you prefer.

Reply:
I updated the code to not always allow to have named arguments. It'll fail earlier in this case.

However, I still think we should keep the `named_arguments` set checks for duplicated names as it makes the errors much better (otherwise we don't get the origin of the error since we don't have the template text anymore when generating the code).